### DISCUSSION TITLE
simple example for trame on django

### DISCUSSION THREAD

**Discussion by geacomputing** [Kitware/trame#395] [Trame/ParaView]
Dear people, hello all. 

I came across trame and is simply MIND BLOWING. What a fantastic job you did. 

I wander if you could provide a simple running example illustrating how trame could be implemented inside DJANGO. I have some data, in vtk form, I would be interested in showing using trame on my webpage. 

I am working with numerical modelling, in this specific case oceanographic model, and vertical coordinate are so called "terrain following". Paraview splendidly demonstrates that and I'd love to implement trame instead of the same, ugly and boring static pictures. 

Data in question are volumetric, so using a slider, the user could also slice through the dataset. 

Thank you for considering my request
Bye

**Comment by jourdain** [comment]
Thanks for reaching out. 

Usually with trame and VTK, you want a dedicated process to prevent any locking or possible crash from your main web server. While it is possible to embed a trame within a web server (like django) by relying on [asyncio](https://docs.djangoproject.com/en/5.0/topics/async/), we prefer to deploy it more like a "micro services". Meaning on the side of your main server while allowing it to be embedded within your views.

It might be tricky to go over all the details on that discussion thread or via an example since deployment tends to be specific. Feel free to reach out to [Kitware](https://www.kitware.com/contact/) so we can really understand your setup and provide a proper answer for your needs.

**Reply by geacomputing** [reply]
Hi Jourdain, 

thanks for the feedback and sorry for the not-so-prompt reply, was out of office. 

I tried to follow the link you have posted in you previous message (https://www.kitware.com/contact/) and that go me here again. So, I hope it is ok writing here. 

I have some vtk files, I generated by converting my numerical model data (from netcdf to vtk) and I would like to add a visualization facility (dedicated section) to my already existing/running webpage, to illustrate and demonstrate the amazing tool that trame is, as well as serving as showcase. 

Nothing really complicated, a simple, fairly light-weighted, vtk file stored on the server side (in my case: heroku) used to show model results. I was hoping to find a template example for django (that is the framework I am currently using), and adapt it for my case. 

Is there something like this, available out there? 

Thank you so much 
Kind regards, 
Marco

**Reply by jourdain** [reply]
It seems that you are just looking at a viewer that can be done with vtk.js.
Otherwise, trame can not be supported by heroku due to its stateless scalability approach. Trame requires a persistent websocket connection.

**Comment by geacomputing** [comment]
Ok Jourdain. Thanks for the feedback. Will explore that. Cheers,

**Reply by jourdain** [reply]
Example of vtk.js viewer https://kitware.github.io/glance/
